### PR TITLE
Make runs tolerant to context.log machinery errors for user-generated logs

### PR DESCRIPTION
**Pull request by gibsondan**
Summary:
This is intended to be a lot like @prha's https://github.com/dagster-io/dagster/pull/6122, but for log.info calls instead of compute log calls. If you write a log.info call and it fails, that's worth logging, but probably is not enough to justify failing the entire step and run.

Specifically really large log calls might run into size limits or other issues related to the content of the custom log, and that shouldn't fail the whole run.

Contrast with failures writing the system events, which do meet the bar for failing the run since they will gum up the execution as a whole)

## Checklist

- [ ] My change requires a change to the documentation and I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: python_modules/dagster/dagster_tests/core_tests/test_python_logging.py [hunk 1]
```diff
@@ -13,11 +16,15 @@
 from dagster.core.test_utils import default_mode_def_for_test, instance_for_test
 
 
+@pytest.fixture
```
Comment (OwenKephart) [inline python_modules/dagster/dagster_tests/core_tests/test_python_logging.py:19]: nice

File: python_modules/dagster/dagster/core/instance/__init__.py [hunk 2]
```diff
@@ -1316,7 +1336,13 @@ def report_engine_event(
 
         check.class_param(cls, "cls")
         check.str_param(message, "message")
-        check.inst_param(pipeline_run, "pipeline_run", PipelineRun)
+        check.opt_inst_param(pipeline_run, "pipeline_run", PipelineRun)
+        check.opt_str_param(run_id, "run_id")
+        check.opt_str_param(pipeline_name, "pipeline_name")
+
+        run_id = run_id if run_id else pipeline_run.run_id
```
Comment (OwenKephart) [inline python_modules/dagster/dagster/core/instance/__init__.py:1343]: are we sure that we will never get here with run_id=None and pipeline_run=None? (similar comment for the line below)

File: python_modules/dagster/dagster/core/instance/__init__.py [hunk 3]
```diff
@@ -130,24 +130,42 @@ def __init__(self, instance):
...
 
+        try:
             self._instance.handle_new_event(event)
-
-        except Exception as e:  # pylint: disable=W0703
-            logging.critical("Error during instance event listen")
-            logging.exception(str(e))
-            raise
+        except Exception as e:
+            if "root" in self._instance.managed_python_loggers:
```
Comment (alangenfeld) [inline python_modules/dagster/dagster/core/instance/__init__.py:148]: what exactly does `"root"` imply here? Feels like this is making an assumption that could change over time

**Comment by gibsondan** [comment]
I think we should shoot for getting either this or https://github.com/dagster-io/dagster/pull/6324 or both out in the release tomorrow (this one is probably the more systemic fix)

**Review by alangenfeld — APPROVED** [review]

**Pull request merged**